src/Neo/ProtocolSettings.cs

```diff
@@ -124,6 +124,42 @@ public record ProtocolSettings
 
         public static ProtocolSettings Custom { get; set; }
 
+        /// <summary>
+        /// Searches for a file in the given path. If not found, checks in the executable directory.
+        /// </summary>
+        /// <param name="fileName">The name of the file to search for.</param>
+        /// <param name="path">The primary path to search in.</param>
+        /// <returns>Full path of the file if found, null otherwise.</returns>
+        public static string FindFile(string fileName, string path)
+        {
+
+            // Check if the given path is relative
+            if (!Path.IsPathRooted(path))
+            {
+                // Combine with the executable directory if relative
+                var executablePath = AppDomain.CurrentDomain.BaseDirectory;
+                path = Path.Combine(executablePath, path);
+            }
+
+            // Check if file exists in the specified (resolved) path
+            var fullPath = Path.Combine(path, fileName);
+            if (File.Exists(fullPath))
+            {
+                return fullPath;
+            }
+
+            // Check if file exists in the executable directory
+            var executableDir = AppDomain.CurrentDomain.BaseDirectory;
```

Comment:
AppContent.BaseDirectory
```suggestion
            var executableDir = AppContext.BaseDirectory;
```

This works on all OSs and NetStandard